### ISSUE TITLE
KL Loss

### ISSUE THREAD

**Issue by Wong-P** [qihao067/CrossFlow#10]
kld_loss = -0.5 * torch.sum(1 + log_var - mu ** 2 - log_var.exp(), dim = 1)
kld_loss = -0.5 * torch.sum(1 + log_var - (0.3 * mu) ** 6 - log_var.exp(), dim = 1)
kld_loss_weight = 1e-2 # 0.0005

Here,  increasing the weight is to enhance diversity. But what is the purpose of adjusting mu**2 to (0.3*mu)**6?

**Comment by qihao067** [comment]
Hello,

This is an empirical tweak we found to be helpful in an ablation study on a small model. 

Replacing mu ** 2 with (0.3 * mu) ** 6 reduces the penalty for small values of mu while increasing the penalty for larger values sharply. This introduces a rapidly growing constraint on the mean, encouraging more expressive latent representations without allowing mu to drift excessively. It aims to enhance diversity by promoting latent space utilization while maintaining control over extreme deviations.

**Comment by Wong-P** [comment]
> Hello,
> 
> This is an empirical tweak we found to be helpful in an ablation study on a small model.
> 
> Replacing mu ** 2 with (0.3 * mu) ** 6 reduces the penalty for small values of mu while increasing the penalty for larger values sharply. This introduces a rapidly growing constraint on the mean, encouraging more expressive latent representations without allowing mu to drift excessively. It aims to enhance diversity by promoting latent space utilization while maintaining control over extreme deviations.

**Comment by Wong-P** [comment]
Thanks!
